src/spine.js

```diff
@@ -127,15 +130,37 @@ pc.extend(pc, function () {
 
         this._position = new pc.Vec3();
 
-        var atlas = new spine.TextureAtlas(atlasData, function (path) {
-            return new SpineTextureWrapper(textureData[path]);
-        });
+        var atlas;
+
+        // EP: As instructed: https://github.com/playcanvas/playcanvas-spine/pull/73
+        //
+        // API differs before we can know which which file version this is supposed to support.
+        // The data file opened will determine which code paths are used.  This file may
+        // or may not be concatenated with the spine library that supports it.
+        //
+        // use the length of the function signatures to guess which library has been concatenated.
+        if (spine.TextureAtlas.length === 1) {
+            // spine 4.1
+            atlas = new spine.TextureAtlas(atlasData);
+            for (const page of atlas.pages) {
+                page.setTexture(new SpineTextureWrapper(textureData[page.name]));
+            }
+        } else {
+            // spine 3.6 and 3.8
+            atlas = new spine.TextureAtlas(atlasData, function (path) {
+                return new SpineTextureWrapper(textureData[path]);
+            });
+        }
+
         var json = new spine.SkeletonJson(new spine.AtlasAttachmentLoader(atlas));
         json.scale *= 0.01;
         var _skeletonData = json.readSkeletonData(skeletonData);
+
+        // Compatibility queries
         this.skeletonVersion = _skeletonData.version;
-        this._spine_3_6_0 = versionCompare(this.skeletonVersion, "3.6.0") <= 0;
-        this._spine_3_7_99 = versionCompare(this.skeletonVersion, "3.7.99") <= 0;
+        this._spine_3_6_0 = versionCompare(this.skeletonVersion, "3.6.0") <= 0; // version 3.6.0 or below
+        this._spine_3_7_99 = versionCompare(this.skeletonVersion, "3.7.99") <= 0; // version 3.7.99 or below
+        this._spine_4_1_X = semver.satisfies(semver.valid(semver.coerce(this.skeletonVersion)), '~4.1.23'); // version 4.1 family
```

Comment:
I suspect you're apprehensive about breaking something, but can't we switch all version testing to `semver` and delete `versionCompare`?

Reply:
I am.  Ok, best approximation completed in 13dc31e